packages/tpetra/core/src/Tpetra_Details_PackTraits.hpp

```diff
@@ -165,17 +167,26 @@ struct PackTraits {
   /// \param numEnt [in] Number of entries to pack.
   ///
   /// \return The number of bytes used to pack \c inBuf into \c outBuf.
+  KOKKOS_INLINE_FUNCTION
   static size_t
   packArray (const output_buffer_type& outBuf,
              const input_array_type& inBuf,
-             const size_t numEnt)
+             const size_t numEnt,
+             int& stat,
+             std::unique_ptr<std::string>& errStr)
```

Comment:
`std::unique_ptr` and `std::string` won't work in CUDA.  Just collect an error code at this point, e.g., by returning `Kokkos::pair<size_t, int>` or even `Kokkos::pair<size_t, size_t>`.  Don't turn it into a string until it gets back to host.